// ngx-copypaste/esm2015/app/ngx-copypaste/ngx-copypaste.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxCopyPasteDirective } from './ngx-copypaste.directive';
import * as ɵngcc0 from '@angular/core';
export class NgxCopyPasteModule {
}
NgxCopyPasteModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxCopyPasteModule });
NgxCopyPasteModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxCopyPasteModule_Factory(t) { return new (t || NgxCopyPasteModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxCopyPasteModule, { declarations: function () { return [NgxCopyPasteDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxCopyPasteDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxCopyPasteModule, [{
        type: NgModule,
        args: [{
                declarations: [NgxCopyPasteDirective],
                imports: [
                    CommonModule
                ],
                exports: [NgxCopyPasteDirective]
            }]
    }], null, null); })();
