// ngx-copypaste/esm2015/app/ngx-copypaste/ngx-copypaste.directive.js
import { Directive, ElementRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class NgxCopyPasteDirective {
    constructor(_elementRef) {
        this._elementRef = _elementRef;
    }
    copy() {
        let select = window.getSelection();
        if (select) {
            select.removeAllRanges();
        }
        const element = this._elementRef.nativeElement;
        if (element instanceof HTMLInputElement ||
            element instanceof HTMLTextAreaElement) {
            this._elementRef.nativeElement.select();
        }
        else {
            const range = document.createRange();
            range.selectNodeContents(this._elementRef.nativeElement);
            select = window.getSelection();
            if (select) {
                select.addRange(range);
            }
        }
        document.execCommand('copy');
    }
}
NgxCopyPasteDirective.ɵfac = function NgxCopyPasteDirective_Factory(t) { return new (t || NgxCopyPasteDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
NgxCopyPasteDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NgxCopyPasteDirective, selectors: [["", "ngxCopyPaste", ""]], exportAs: ["copy"] });
NgxCopyPasteDirective.ctorParameters = () => [
    { type: ElementRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxCopyPasteDirective, [{
        type: Directive,
        args: [{
                selector: '[ngxCopyPaste]',
                exportAs: 'copy'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, null); })();
